From: Phil Elwell <phil@raspberrypi.com>
Date: Thu, 2 Jul 2020 15:21:05 +0100
Subject: overlays: Fix miniuart-bt "krnbt" parameter

Although superficially appealing, an overlay parameter that targets a
label in the base DTB is not currently supported. Instead it is
necessary to create a fragment targeting the label which is patched
by the parameter.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts b/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts
index 44d8e10ef735..da49f14a0940 100644
--- a/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts
+++ b/arch/arm/boot/dts/overlays/miniuart-bt-overlay.dts
@@ -81,7 +81,13 @@
 		};
 	};
 
+	fragment@7 {
+		target = <&minibt>;
+		minibt_frag: __overlay__ {
+		};
+	};
+
 	__overrides__ {
-		krnbt = <&minibt>,"status";
+		krnbt = <&minibt_frag>,"status";
 	};
 };
